src/networks/metrics.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::sync::Arc;
+
+use once_cell::sync::Lazy;
+use prometheus_client::metrics::gauge::Gauge;
+
+use super::ChainConfig;
+
+pub static EXPECTED_NETWORK_HEIGHT: Lazy<Gauge> = Lazy::new(|| {
+    let metric = Gauge::default();
+    crate::metrics::default_registry().register(
+        "expected_network_height",
+        "The expected network height based on the current time and the genesis block time.",
+        metric.clone(),
+    );
+    metric
+});
+
+/// Task to periodically update network-related metrics.
+pub async fn network_metrics_loop(config: Arc<ChainConfig>, genesis_timestamp: u64) {
+    loop {
+        // For this particular metric, we theoretically could have initiated the value once and
+        // increased it by one every `block_delay_secs`, but this would have made the metric less
+        // responsive and more fragile, e.g., on system clock changes. Given that the operation is cheap, we just
+        // recalculate it every second.
+        let now_epoch = chrono::Utc::now()
```

Comment:
We have a `fn get_now_epoch` in `chain_muxer.rs`, maybe we could move around and re-use it